C++ API parity: RReLU

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* #27537 C++ API parity: Tanhshrink
* #27536 C++ API parity: Tanh
* #27535 C++ API parity: Softsign
* #27534 C++ API parity: Softshrink
* #27489 C++ API parity: Softplus
* #27488 C++ API parity: Sigmoid
* #27487 C++ API parity: CELU
* **#27437 C++ API parity: RReLU**

Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

pbelevich committed:
C++ API parity: RReLU

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"




[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -57,6 +57,14 @@ inline Tensor relu6(Tensor& input, const ReLU6Options& options) {
     HardtanhOptions().min_val(0).max_val(6).inplace(options.inplace()));
 }
 
+inline Tensor rrelu(Tensor& input, const RReLUOptions& options, bool training=false) {
```
similarly here: `options` can be optional, and we should add a test for the no options case too.

pbelevich committed:
Update on "C++ API parity: RReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"




[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -57,6 +57,15 @@ inline Tensor relu6(Tensor& input, const ReLU6Options& options = {}) {
 }
 
+inline Tensor rrelu(Tensor& input, const RReLUOptions& options = {},
+                    bool training=false) {
```
nit:
```suggestion
                    bool training = false) {
```

yf225 reviewed (approved):
Thanks a lot for the great work @pbelevich !

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: RReLU"


Differential Revision: [D17835413](https://our.internmc.facebook.com/intern/diff/D17835413)

[ghstack-poisoned]
